File: src/main/java/io/playground/orderservice/application/order/dto/OrderDto.java
package io.playground.orderservice.application.order.dto;

import java.math.BigDecimal;
import java.util.Map;

public class OrderDto {
    public record DoOrderRequest(
            String orderExternalId,
            Map<Long, Integer> variantQuantities
    ) {
    }

    public record DoPaymentRequest(
            String idempotencyKey,
            String orderExternalId,
            String paymentKey,
            BigDecimal amount
    ) {
    }

    public record CancelAllRequest(
            String idempotencyKey,
            String orderExternalId,
            String paymentKey,
            String reason
    ) {
    }

    public record CancelPartiallyRequest(
            String idempotencyKey,
            String orderExternalId,
            String paymentKey,
            Map<Long, Integer> variantQuantities,
            String reason
    ) {
    }
}

File: src/main/kotlin/io/playground/orderservice/application/order/dto/OrderDto.kt
package io.playground.orderservice.application.order.dto

import kotlin.jvm.JvmRecord

class OrderDto {
    @JvmRecord
    data class DoOrderRequest(
        val orderExternalId: String,
        val variantQuantities: Map<Long, Int>,
    )

    @JvmRecord
    data class DoPaymentRequest(
        val idempotencyKey: String,
        val orderExternalId: String,
        val paymentKey: String,
        val amount: java.math.BigDecimal,
    )

    @JvmRecord
    data class CancelAllRequest(
        val idempotencyKey: String,
        val orderExternalId: String,
        val paymentKey: String,
        val reason: String,
    )

    @JvmRecord
    data class CancelPartiallyRequest(
        val idempotencyKey: String,
        val orderExternalId: String,
        val paymentKey: String,
        val variantQuantities: Map<Long, Int>,
        val reason: String,
    )
}
